engine/src/test/java/org/pentaho/di/trans/steps/databasejoin/DatabaseJoinTest.java

```diff
@@ -123,4 +132,82 @@ public void testStopRunningWhenStepIsNotStoppedNorStepDataInterfaceIsDisposedAnd
     verify( mockStepDataInterface.db, times( 0 ) ).cancelStatement( any( PreparedStatement.class ) );
     assertFalse( mockStepDataInterface.isCanceled );
   }
+
+  @Test
+  public void testLookupStopRunningPossibleDeadlock() throws Exception {
```

Comment:
We ran this test multiple times without the fix above i.e. against the old code, and it never failed.